src/components/Content/index.tsx

```diff
@@ -11,32 +11,35 @@ import { routes } from "constants/routes";
 import { FullPageLoad } from "components/Loading/FullPageLoad";
 import { useAuthStateContext } from "context/auth";
 import { useQuery } from "@apollo/react-hooks";
-import { GET_PROJECTS } from "gql/queries/get-projects";
-import { ProjectsQuery } from "gql/generated/types";
+import { GET_USER } from "gql/queries";
+import { GetUserQuery } from "gql/generated/types";
 import { PageLayout } from "components/styles/Layout";
 import { PageDoesNotExist } from "pages/404";
 import { ConfigurePatch } from "pages/ConfigurePatch";
 import { Preferences } from "pages/Preferences";
+import { MyPatches } from "pages/MyPatches";
 
 export const Content: React.FC = () => {
   const { isAuthenticated, initialLoad } = useAuthStateContext();
 
   // this top-level query is required for authentication to work
   // afterware is used at apollo link level to authenticate or deauthenticate user based on response to query
   // therefore this could be any query as long as it is top-level
-  useQuery<ProjectsQuery>(GET_PROJECTS);
+  useQuery<GetUserQuery>(GET_USER);
 
   if (!isAuthenticated && initialLoad) {
     return <FullPageLoad />;
   }
+
   return (
     <PageLayout>
       <Navbar />
       <Switch>
         <PrivateRoute path={routes.task} component={Task} />
         <PrivateRoute path={routes.configurePatch} component={ConfigurePatch} />
         <PrivateRoute path={routes.patch} component={Patch} />
-        <PrivateRoute path={routes.myPatches} component={MyPatches} />
+        <PrivateRoute exact path={routes.myPatches} component={MyPatches} />
```

Comment:
I opted to create a component that redirects than do pass in userId from the GET_USER query because waiting for a valid response object before rendering these routes prevents PrivateRoute from redirecting to login when the user is not logged in

Reply:
the loading screen flashes and then goes away even when the user is not authenticated because of the "initialLoad" variable in the condition. 
The user is redirected to the login screen because we allow PrivateRoutes to mount and redirect the user even when they are not authenticated and when the result of the user data query is undefined. 

doing something like this doesn't work because the route does not update when userId has a value. 
```
<PrivateRoute exact path={routes.myPatches}>
   <Redirect to={`${paths.user}/${get(data, "user.userId")}/patches`} />
</PrivateRoute>
```

if we block mounting the routes when result of the query isn't defined, the user will never be be redirected to the login page.

Reply:
okay, makes sense